### PR TITLE
Postmotpost and reskontro voucher print fix

### PR DESCRIPTION
**Pull request by mladjo2505**
An old task.
Can be found on trello. https://trello.com/c/p0rfBcBc/419-lodo-postmotpost-report-reconciling-md-fix2

### CODE REVIEW

**Comment by stoivo** [comment]
Rebase to remove all merge commits.
Add link to trello task.

**Comment by mladjo2505** [comment]
@stoivo Added link
Rebasing with preserving the merge commits is something I have never done, so I wouldn't risk it without researching it a lot. You are welcome to try it yourself and show me how you did it.

**Comment by stoivo** [comment]
I can do it with you.

File: db/changes/111_fix_voucherstruct_postmotpost_data.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+UPDATE voucher v,
+(
+SELECT vs.ParentVoucherID, vs.ChildVoucherID,
+v1.VoucherID as VoucherID1, v1.InvoiceID as InvoiceID1, v1.KID as KID1, vm1.MatchNumber as MatchNumber1, v1.matched_by as matched_by1,
+v2.VoucherID as VoucherID2, v2.InvoiceID as InvoiceID2, v2.KID as KID2, vm2.MatchNumber as MatchNumber2, v2.matched_by as matched_by2,
+IF(v1.InvoiceID = v2.InvoiceID AND v1.InvoiceID <> '' AND v1.InvoiceID IS NOT NULL AND v2.InvoiceID <> '' AND v2.InvoiceID IS NOT NULL, 'true', 'flase') as match_invoice,
+IF(v1.KID = v2.KID AND v1.KID <> '' AND v1.KID IS NOT NULL AND v2.KID <> '' AND v2.KID IS NOT NULL, 'true', 'false') as match_kid,
+IF(vm1.MatchNumber = vm2.MatchNumber AND vm1.MatchNumber <> 0 AND vm1.MatchNumber IS NOT NULL AND vm2.MatchNumber <> 0 AND vm2.MatchNumber IS NOT NULL, 'true', 'flase') as match_matchnumber
+FROM voucherstruct vs, voucher v1, voucher v2, vouchermatch vm1, vouchermatch vm2
+WHERE vs.ParentVoucherID = v1.VoucherID AND vs.ChildVoucherID = v2.VoucherID AND vs.ParentVoucherID = vm1.VoucherID AND vs.ChildVoucherID = vm2.VoucherID AND (v1.matched_by = 0 OR v2.matched_by = 0)
```
Comment (mladjo2505) [inline db/changes/111_fix_voucherstruct_postmotpost_data.sql:14]: What if matched_by is an empty string? Maybe write a migration to set all empty string values for matched_by to 0.

File: modules/postmotpost/model/postmotpost.class.php [hunk 2]
```diff
@@ -1033,5 +1055,16 @@ function findMotKonto($AccountPlanID) {
 
         return $_lib['db']->db_fetch_assoc($r);
     }
+
+    private function updateClosedWith($VoucherID, $VoucherID2, $ClosedWith) {
+        global $_lib;
+
+        $args = array("voucher_matched_by_$VoucherID" => $ClosedWith, "voucher_matched_by_$VoucherID2" => $ClosedWith);
+        $pk = array('voucher' => 'VoucherID');
+        $_lib['db']->db_update_hash($args, 'voucher', array("CarID" => $CarID, "MilageYear" => $milage_year));
```
Comment (mladjo2505) [inline modules/postmotpost/model/postmotpost.class.php]: This line should not be here

**Comment by sls-lnluksa** [comment]
This branch needs to be tested by Hildegunn or someone who is using LODO, before it goes on live server.

**Comment by stoivo** [comment]
Ok, we can deploy it to lodo2016.

**Comment by stoivo** [comment]
Check why report.reskontrovoucherprint Diff is printed wrong.

File: modules/report/view/reskontrovoucherprint.php [hunk 3]
```diff
@@ -57,6 +58,13 @@
     $whereouter .= " v.DepartmentID = " . $_REQUEST['report_DepartmentID'] . " and ";
 }
 
+if ( in_array($selectedAccount->AccountPlanID, array(1500, 2400, 2930)))
+{
+  if ($selectedAccount->AccountPlanID == 1500) $whereouter .= " a.AccountPlanType = 'customer' and ";
+  if ($selectedAccount->AccountPlanID == 2400) $whereouter .= " a.AccountPlanType = 'supplier' and ";
+  if ($selectedAccount->AccountPlanID == 2930) $whereouter .= " a.AccountPlanType = 'employee' and ";
+}
+
```
Comment (mladjo2505) [inline modules/report/view/reskontrovoucherprint.php:60]: Instead of these 3 ifs we should use `$selectedAccount->ReskontroAccountPlanType`

**Pull request merged**